### ISSUE TITLE
Option to select all connected features/elements

### ISSUE THREAD

**Issue by forteller** [openstreetmap/iD#7388]
I spend maybe too much time adjusting the placement of buildings. This might not be an important thing to do, generally. But some times I have to to make sure the building doesn't cross a road or a sidewalk. But this is not a fun thing to do if the building is a part of a complex with many connected buildings. Especially if I miss one of the connected element, starts moving it, notice it distort the element I missed, and have to start all over again.

It would be very helpful to have a way to click on one element and then automatically select all connected elements. Maybe as a keyboard shortcut, or maybe as something in the right click context menu. That's not so important for me.

Thanks!

![Screenshot from 2020-02-26 21-05-19](https://user-images.githubusercontent.com/343561/75385730-92fa0980-58e0-11ea-9289-c499eb6f9db9.png)

(Example of a building crossing a sidewalk because it, and all its connected buildings are misaligned.)

**Comment by quincylvania** [comment]
I agree we should improve iD's selection mechanisms since we've been improving multiselection inspection and editing for v3. We'll have to see if we can figure out an intuitive UI for this. See also #7375.

**Comment by forteller** [comment]
Thanks! The lasso tool is not very relevant, though, since it also marks all freestanding nodes etc in the area. As you can see in the screenshot there's a lot of those in Norway (all addresses are added in this way), and you often don't want to move this. In the lower right corner of the screenshot you can see buildings that should be moved, but if the address nodes are moved just as much as the buildings then the nodes would be moved outside the buildings in the photo.

**Comment by quincylvania** [comment]
#7375 is distinct but might be helpful in some of these cases, such as when separate address points aren't present.

**Comment by forteller** [comment]
Hah, typical of me to not learn about #8264 until now :P            

Still, this issue is still wanted, and I want to suggest a shortcut for it: So when Ctrl+up selects all nodes of the current feature, what if hitting up again after that then selected all connected nodes? Seems like a logical way to do it.

**Comment by 1ec5** [comment]
Ctrl+Up should already select all the ways connected to the selected nodes, while Ctrl+Down selects all the nodes connected to those ways. You can wind up with a pretty large selection by alternating these two shortcuts. 😉 This feature only navigates among nodes and ways, not relations and their members. That would be a handy feature too, but it would probably be a bit more complicated due to sparse relations and such.
